rllib/policy/sample_batch.py

```diff
@@ -103,6 +103,9 @@ def __init__(self, *args, **kwargs):
         self.zero_padded = kwargs.pop("_zero_padded", False)
         # Whether this batch is used for training (vs inference).
         self._is_training = kwargs.pop("_is_training", None)
+        # Average number of grad updates that have been performed
```

Comment:
is it really the average number of updates? I think its the total sum of gradient updates that have occurred over the life of the trained policy.

We need to update this comment, unless you intended to say "Average" in which case needs to be reflected in the code.

Reply:
Good question. So it's actually the weighted average over the different samples (timesteps) in that batch.

Imagine you have two rollout workers and one of them has a policy that has undergone 2 updates thus far, the other uses a policy that has undergone 3 updates thus far. Both workers produce rollouts of length 50 (rollout_fragment_length==50). The train batch size is 100, so we concatenate these 2x len=50 batches to one that's 100 ts long. This new 100ts batch will have its `num_gradient_updates` property set to 2.5 as it's the weighted average (both original batches contribute 50%).

Let me try making this more clear in the comment here ..

Reply:
Enhanced the comment to make this more clear.

Reply:
oh I see. sg